stm/stm.ml

```diff
@@ -191,9 +191,9 @@ let mkTransCmd cast cids ceff cqueue =
 
 type cached_state =
   | EmptyState
-  | ParsingState of Vernacstate.Parser.t
+  | ParsingState of Pcoq.frozen_t
```

Comment:
It would have been nice to have here `Vernacstate.Synterp.t`, however it doesn't work.

Reply:
what breaks exactly?

Reply:
IIRC the breakage was big.

Reply:
Yes:
```
File "./theories/Init/Notations.v", line 117, characters 0-16:
Error: There is nothing to end.
```
Maybe it is something easy, but I don't have the cycles to debug the Stm (and likely vsCoq).

Reply:
Branch at https://github.com/ejgallego/coq/tree/stm_use_synterp_state

Reply:
probably because Module s classified VtLater
I guess VtNow vs VtLater should be replaced by checking if it's VernacSynPure

Reply:
Anything that goes towards removing the `vernac_classify` stuff is IMHO worth it.